src/coreclr/jit/importer.cpp

```diff
@@ -18930,41 +18930,42 @@ void Compiler::impMakeDiscretionaryInlineObservations(InlineInfo* pInlineInfo, I
     GenTreeCall::Use* argUse = pInlineInfo == nullptr ? nullptr : pInlineInfo->iciCall->AsCall()->gtCallArgs;
     for (unsigned i = 0; i < info.compMethodInfo->args.numArgs; i++)
     {
-        CORINFO_CLASS_HANDLE sigClass = info.compCompHnd->getArgClass(&sig, sigArg);
-        GenTree*             argNode  = argUse == nullptr ? nullptr : argUse->GetNode()->gtSkipPutArgType();
+        CORINFO_CLASS_HANDLE sigClass;
+        CorInfoType          corType = strip(info.compCompHnd->getArgType(&sig, sigArg, &sigClass));
+        sigClass                     = info.compCompHnd->getArgClass(&sig, sigArg);
```

Comment:
For non-ref-class typed args doesn't this trash `sigClass`? Seems like you'd want something like:
```suggestion
        sigClass                     = (corType == CORINFO_TYPE_CLASS) ? info.compCompHnd->getArgClass(&sig, sigArg) : sigClass;
```
At least in `corinfo.h`, `getArgClass` is not documented to do anything useful for non-ref classes.

Reply:
Fixed, thanks! However, it's also needed for byref references, e.g. `ref string arg` - in order to get CLS for string we have to do `getArgClass` and `getChildType`